cpp/src/io/parquet/compact_protocol_reader.cpp

```diff
@@ -399,6 +404,7 @@ struct parquet_field_binary_list
       auto const l = cpr->get_u32();
       CUDF_EXPECTS(l <= static_cast<size_t>(cpr->m_end - cpr->m_cur), "binary length mismatch");
 
+      CUDF_EXPECTS(i >= 0 and i < val.size());
```

Comment:
Looks like the vector is resized in the next line so maybe this check should include `l` instead of `val.size()`